kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/internal/CertUtilsTest.java

```diff
@@ -184,6 +184,50 @@ void getInputStreamFromDataOrFileShouldDecodeBase64EncodedString() throws IOExce
     assertEquals(inputStr, certDataReadFromInputStream);
   }
 
+  @Test
+  void loadECkeys()
+      throws InvalidKeySpecException, CertificateException, NoSuchAlgorithmException, KeyStoreException, IOException {
+    String privateKeyPath = Utils.filePath(getClass().getResource("/ssl-test/fabric8-ec.paired.key"));
+    String certPath = Utils.filePath(getClass().getResource("/ssl-test/fabric8-ec.cert"));
+
+    KeyStore trustStore = CertUtils.createKeyStore(null, certPath, null, privateKeyPath, "EC", "foo", null, null);
+
+    assertEquals(1, trustStore.size());
+  }
+
+  @Test
+  void loadECPrivateOnlyKey()
+      throws InvalidKeySpecException, CertificateException, NoSuchAlgorithmException, KeyStoreException, IOException {
+    String privateKeyPath = Utils.filePath(getClass().getResource("/ssl-test/fabric8-ec.private-only.key"));
+    String certPath = Utils.filePath(getClass().getResource("/ssl-test/fabric8-ec.cert"));
+
+    KeyStore trustStore = CertUtils.createKeyStore(null, certPath, null, privateKeyPath, "EC", "foo", null, null);
+
+    assertEquals(1, trustStore.size());
+  }
+
+  @Test
+  void loadNothingError()
+      throws InvalidKeySpecException, CertificateException, NoSuchAlgorithmException, KeyStoreException, IOException {
+    String privateKeyPath = Utils.filePath(getClass().getResource("/ssl-test/empty"));
+    String certPath = Utils.filePath(getClass().getResource("/ssl-test/empty"));
+
+    Exception exception = assertThrows(KubernetesClientException.class,
+        () -> CertUtils.createKeyStore(null, certPath, null, privateKeyPath, "EC", "foo", null, null));
+    assertTrue(exception.getMessage().contains("Could not load EC key."));
+  }
+
+  @Test
+  void loadUnknownError() {
+    String privateKeyPath = Utils.filePath(getClass().getResource("/ssl-test/multiple-certs.p7b"));
+    String certPath = Utils.filePath(getClass().getResource("/ssl-test/multiple-certs.p7b"));
+
+    Exception exception = assertThrows(KubernetesClientException.class,
+        () -> CertUtils.createKeyStore(null, certPath, null, privateKeyPath, "EC", "foo", null, null));
+    assertTrue(exception.getMessage().contains("Don't know what to do with a"));
```

Comment:
We're also using assertj in this test. This would become more readable with assertj:
```java
assertThatExceptionOfType(KubernetesClientException.class)
  .isThrownBy(() -> CertUtils.createKeyStore(null, certPath, null, privateKeyPath, "EC", "foo", null, null))
  .withMessageContaining("Don't know what to do with a");
```